packages/labs/nextjs/src/index.ts

```diff
@@ -0,0 +1,85 @@
+/**
+ * @license
+ * Copyright 2023 Google LLC
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import type {NextConfig} from 'next';
+
+// TODO(augustjk) Consider adding option for adding template shadowroot polyfill
+// automatically
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
+interface LitSsrPluginOptions {}
+
+export = (_pluginOptions: LitSsrPluginOptions = {}) =>
+  (nextConfig: NextConfig = {}) => {
+    return Object.assign({}, nextConfig, {
+      webpack: (config, options) => {
+        const {isServer, nextRuntime, webpack} = options;
+        config.module.rules.unshift({
+          // Grab entry points for all pages
+          // TODO(augustjk) This may not work for the new "app" directory
+          // introduced in Next.js v13
+          test: /\/pages\//,
+          // Exclude Next's own distributed files as they're commonjs and won't
+          // play nice with `imports-loader`
+          exclude: /next\/dist\//,
+          loader: 'imports-loader',
+          options: {
+            imports: ['side-effects @lit-labs/ssr-react/enable-lit-ssr.js'],
```

Comment:
Maybe a few comments just to describe high level what's going on?

I think this line means that we'll load the module that monkey patches React.createElement?